### ISSUE TITLE
More configurable `ignorePatterns`.

### ISSUE THREAD

**Issue by GeorgySerga** [eslint/eslint#12917]
**The version of ESLint you are using.**
6.1.0

**The problem you want to solve.**
`ignorePatterns` set though configuration file. Currently it accepts an array of ignored paths and doesn't allow to use existing ignore configurations. (link to RFC: https://github.com/eslint/rfcs/pull/22/files#diff-ceb32bac770d76f9f72f2b2667bd016bR79)

**Your take on the correct solution to problem.**
To accept either a string with ignore file name, so it would resemble cli option, i.e. `'.gitignore'` (less flexible, easier to implement), or an object (more flexible and harder to implement):
```
{
  patterns: ['file.js'], // same as now
  ignoreSets: ['.gitignore'] // files to import, like the cli option. It's an array to make it extra flexible, multiple files can be specified if people want to
}
```

Judging by the comments in the issues referenced in the RFC people wanted a configuration option to make eslintrc a more unified place, reusing existing configurations (like `.gitignore`).

**Are you willing to submit a pull request to implement this change?**
Yes, I can work on this if the general idea is interesting (I'm not part of eslint community, maybe this idea isn't aligned with project). A bit hesitant about the naming in the object example above. Would love any suggestions.

Thank you!

**Comment by kaicataldo** [comment]
Since this is a core change, we ask that you create an [RFC](https://github.com/eslint/rfcs) for this so that we can evaluate the proposal. Thanks!

**Comment by mysticatea** [comment]
I think that you can do so with `fs` module.

**.eslintrc.js**:

```js
const fs = require("fs")

module.exports = {
    ignorePatterns: fs.readFileSync("path/to/a/.gitignore").split("\n"),
}
```

**Comment by kaicataldo** [comment]
Oh, good call!

**Comment by GeorgySerga** [comment]
Thank you @mysticatea , good idea. I think this solves it for me. The proposal I've written above is a bit more flexible, but I don't think there's a need. Would close this issue, possible feature for the future if there's a demand, but for now KISS :)

Thank you @kaicataldo as well.
